asmcomp/cmmgen.ml

```diff
@@ -300,12 +300,50 @@ let invert_then_else = function
   | Then_false_else_true -> Then_true_else_false
   | Unknown -> Unknown
 
+let rec simplify_cond = function
+  | Cop(Ccmpi cmpop
+       , [ Cop(Caddi,[Cop(Clsl,[left;Cconst_int(1,_)],_);Cconst_int(1,_)],_)
+         ; Cconst_int (right_i,right_dbg)
+         ]
+       , dbg_cmp
+       ) ->
+      Cop (Ccmpi cmpop, [ left; Cconst_int (right_i asr 1, right_dbg)], dbg_cmp)
+      |> simplify_cond
```

Comment:
Please could you rewrite this to remove the pipe!

Reply:
Yes, the next version will have correct simplifications and no pipes.